File: C#/0119. Pascal's Triangle II.cs
public class Solution {
    public IList<int> GetRow(int rowIndex) {   
        IList<int> nums=new List<int>();
        nums.Add(1);
        if (rowIndex==0){
            return nums;
        }
        nums.Add(1);
        
        IList<int> prenums=nums;
        for(int i=2;i<rowIndex+1;i++){
            IList<int> curnums=new List<int>{1};
            for(int j=0;j<prenums.Count-1;j++){
                curnums.Add(prenums[j]+prenums[j+1]);
            }
            curnums.Add(1);
            prenums=curnums;
        }
        
        return prenums;
    }
}

File: archives/20190519python/0119. Pascal's Triangle II.py
class Solution:
    def getRow(self, rowIndex: int) -> List[int]:
        if rowIndex==0:
            return [1]
        if rowIndex==1:
            return [1,1]
        
        prev=[1,1]
        while(rowIndex>1):            
            cur=[1]
            for k in range(len(prev)-1):
                cur.append(prev[k]+prev[k+1])
            cur.append(1)
            prev=cur
            rowIndex-=1
        return prev
            
